src/Components/CreateImageWizard/CreateImageWizard.js

```diff
@@ -195,9 +195,118 @@ const onSave = (values) => {
+    if (composeRequest) {
+        const imageRequest = composeRequest.image_requests[0];
+        const uploadRequest = imageRequest.upload_request;
+        let formState = {};
+
+        formState['image-name'] = composeRequest.image_name;
+
+        formState.release = composeRequest?.distribution;
+        // set defaults for target environment first
+        formState['target-environment'] = {
+            aws: false,
+            azure: false,
+            gcp: false,
+            'guest-image': false,
+        };
+        // then select the one from the request
+        // if the image type is to a cloud provider we use the upload_request.type
+        // or if the image is intended for download we use the image_type
+        let targetEnvironment;
+        if (uploadRequest.type === 'aws.s3') {
+            targetEnvironment = imageRequest.image_type;
+        } else {
+            targetEnvironment = uploadRequest.type;
+        }
+
+        formState['target-environment'][targetEnvironment] = true;
+
+        if (targetEnvironment === 'aws') {
+            formState['aws-account-id'] = uploadRequest?.options?.share_with_accounts[0];
+        } else if (targetEnvironment === 'azure') {
+            formState['azure-tenant-id'] = uploadRequest?.options?.tenant_id;
+            formState['azure-subscription-id'] = uploadRequest?.options?.subscription_id;
+            formState['azure-resource-group'] = uploadRequest?.options?.resource_group;
+        } else if (targetEnvironment === 'gcp') {
+            // parse google account info
+            // roughly in the format `accountType:accountEmail`
+            const accountInfo = uploadRequest?.options?.share_with_accounts[0];
+            const [ accountTypePrefix, account ] = accountInfo.split(':');
+
+            switch (accountTypePrefix) {
+                case 'user':
+                    formState['google-account-type'] = 'googleAccount';
+                    formState['google-email'] = account;
+                    break;
+                case 'serviceAccount':
+                    formState['google-account-type'] = 'serviceAccount';
+                    formState['google-email'] = account;
+                    break;
+                case 'group':
+                    formState['google-account-type'] = 'googleGroup';
+                    formState['google-email'] = account;
+                    break;
+                case 'domain':
+                    formState['google-account-type'] = 'domain';
+                    formState['google-domain'] = account;
+                    break;
+            }
+        }
+
+        // customizations
+        // packages
+        // TODO: add package description
+        formState['selected-packages'] = composeRequest?.customizations?.packages?.map(packageName => {
+            return ({
+                name: packageName
+            });
+        });
+
+        // filesystem
+        // TODO: intelligent unit selection
```

Comment:
https://github.com/RedHatInsights/image-builder-frontend/commit/6aed45346eb495e85223029cb48eab1099a4ad43#diff-dbd247166ab10488d571ec72c6213aab6fe73f84f43538a34c4e972ce418b738R24-R36 could do something similar to this?

Reply:
This is fixed.